packages/assets-controllers/src/NftController.ts

```diff
@@ -512,6 +512,53 @@ export class NftController extends BaseController<
     });
   }
 
+  /**
+   * Fetches NFT Collection Metadata from the NFT API.
+   *
+   * @param contractAddresses - The contract addresses of the NFTs.
+   * @param chainId - The chain ID of the network where the NFT is located.
+   * @returns NFT collections metadata.
+   */
+  async fetchNftCollectionsMetadata(
+    contractAddresses: string[],
+    chainId: Hex,
+  ): Promise<{
+    collections: Collection[];
+  }> {
+    return await this.#getNftCollectionInformationFromApi(
+      contractAddresses,
+      chainId,
+    );
+  }
+
+  async #getNftCollectionInformationFromApi(
+    contractAddress: string[],
+    chainId: Hex,
+  ) {
+    const urlParams = new URLSearchParams({
+      chainId,
+    });
+
+    for (const address of contractAddress) {
+      urlParams.append('contract', address);
+    }
+
+    return await handleFetch(
+      `${this.#getNftCollectionApi()}?${urlParams.toString()}`,
```

Comment:
`handleFetch` was changed some time ago to take a `URL` object, so instead of constructing the URL like this, what do you think about:

``` suggestion
    const url = new URL(this.#getNftCollectionApi());
    url.searchParams.append('chainId', chainId);

    for (const address of contractAddress) {
      url.searchParams.append('contract', address);
    }

    return await handleFetch(
      url,

Reply:
Done in https://github.com/MetaMask/core/pull/4506/commits/09bfcf5ae6862e55860d7c0cc3fa4541ef491840